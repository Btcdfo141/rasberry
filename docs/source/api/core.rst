.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: