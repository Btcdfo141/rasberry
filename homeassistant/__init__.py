"""Init file for Home Assistant."""


test2
