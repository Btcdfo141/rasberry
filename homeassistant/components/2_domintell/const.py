"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
