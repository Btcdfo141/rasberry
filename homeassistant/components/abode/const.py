"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
