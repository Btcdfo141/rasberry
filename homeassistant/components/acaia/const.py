"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
