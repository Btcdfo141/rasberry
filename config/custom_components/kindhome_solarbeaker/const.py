from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
