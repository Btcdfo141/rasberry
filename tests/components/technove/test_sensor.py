"""Tests for the TechnoVE sensor platform."""
from unittest.mock import MagicMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import STATE_UNKNOWN
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from tests.common import MockConfigEntry


@pytest.mark.usefixtures("entity_registry_enabled_by_default", "mock_technove")
async def test_sensors(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test the creation and values of the TechnoVE sensors."""
    mock_config_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()
    entity_registry = er.async_get(hass)
    entity_entries = er.async_entries_for_config_entry(
        entity_registry, mock_config_entry.entry_id
    )

    assert entity_entries
    for entity_entry in entity_entries:
        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
        assert hass.states.get(entity_entry.entity_id) == snapshot(
            name=f"{entity_entry.entity_id}-state"
        )


@pytest.mark.parametrize(
    "entity_id",
    (
        "sensor.technove_station_signal_strength",
        "sensor.technove_station_wi_fi_network_name",
    ),
)
@pytest.mark.usefixtures("init_integration")
async def test_disabled_by_default_sensors(
    hass: HomeAssistant, entity_registry: er.EntityRegistry, entity_id: str
) -> None:
    """Test the disabled by default TechnoVE sensors."""
    assert hass.states.get(entity_id) is None

    assert (entry := entity_registry.async_get(entity_id))
    assert entry.disabled
    assert entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_no_wifi_support(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_technove: MagicMock,
) -> None:
    """Test missing Wi-Fi information from TechnoVE device."""
    # Remove Wi-Fi info
    device = mock_technove.update.return_value
    device.info.network_ssid = None

    # Setup
    mock_config_entry.add_to_hass(hass)
    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert (state := hass.states.get("sensor.technove_station_wi_fi_network_name"))
    assert state.state == STATE_UNKNOWN
