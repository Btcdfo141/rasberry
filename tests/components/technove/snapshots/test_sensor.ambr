# serializer version: 1
# name: test_sensors[binary_sensor.technove_station_battery_protected-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_battery_protected',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Battery protected',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'is_battery_protected',
    'unique_id': 'AA:AA:AA:AA:AA:BB_is_battery_protected',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_battery_protected-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'TechnoVE Station Battery protected',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_battery_protected',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_sensors[binary_sensor.technove_station_charging-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_charging',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.BATTERY_CHARGING: 'battery_charging'>,
    'original_icon': None,
    'original_name': 'Charging',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'AA:AA:AA:AA:AA:BB_is_session_active',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_charging-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'battery_charging',
      'friendly_name': 'TechnoVE Station Charging',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_charging',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_sensors[binary_sensor.technove_station_conflict_with_power_sharing_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_conflict_with_power_sharing_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Conflict with power sharing mode',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'conflict_in_sharing_config',
    'unique_id': 'AA:AA:AA:AA:AA:BB_conflict_in_sharing_config',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_conflict_with_power_sharing_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'TechnoVE Station Conflict with power sharing mode',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_conflict_with_power_sharing_mode',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_sensors[binary_sensor.technove_station_power_sharing_mode-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_power_sharing_mode',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Power sharing mode',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'in_sharing_mode',
    'unique_id': 'AA:AA:AA:AA:AA:BB_in_sharing_mode',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_power_sharing_mode-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'TechnoVE Station Power sharing mode',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_power_sharing_mode',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_sensors[binary_sensor.technove_station_static_ip-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_static_ip',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Static IP',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'is_static_ip',
    'unique_id': 'AA:AA:AA:AA:AA:BB_is_static_ip',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_static_ip-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'TechnoVE Station Static IP',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_static_ip',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_sensors[binary_sensor.technove_station_update-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'binary_sensor.technove_station_update',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.UPDATE: 'update'>,
    'original_icon': None,
    'original_name': 'Update',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'AA:AA:AA:AA:AA:BB_update_available',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[binary_sensor.technove_station_update-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'update',
      'friendly_name': 'TechnoVE Station Update',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.technove_station_update',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_sensors[sensor.technove_station_current-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_current',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.CURRENT: 'current'>,
    'original_icon': None,
    'original_name': 'Current',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'AA:AA:AA:AA:AA:BB_current',
    'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
  })
# ---
# name: test_sensors[sensor.technove_station_current-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'current',
      'friendly_name': 'TechnoVE Station Current',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_current',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '23.75',
  })
# ---
# name: test_sensors[sensor.technove_station_input_voltage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_input_voltage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.VOLTAGE: 'voltage'>,
    'original_icon': None,
    'original_name': 'Input voltage',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'voltage_in',
    'unique_id': 'AA:AA:AA:AA:AA:BB_voltage_in',
    'unit_of_measurement': <UnitOfElectricPotential.VOLT: 'V'>,
  })
# ---
# name: test_sensors[sensor.technove_station_input_voltage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'voltage',
      'friendly_name': 'TechnoVE Station Input voltage',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfElectricPotential.VOLT: 'V'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_input_voltage',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '238',
  })
# ---
# name: test_sensors[sensor.technove_station_last_session_energy_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_last_session_energy_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.ENERGY: 'energy'>,
    'original_icon': None,
    'original_name': 'Last session energy usage',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'energy_session',
    'unique_id': 'AA:AA:AA:AA:AA:BB_energy_session',
    'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
  })
# ---
# name: test_sensors[sensor.technove_station_last_session_energy_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'energy',
      'friendly_name': 'TechnoVE Station Last session energy usage',
      'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
      'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_last_session_energy_usage',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '12.34',
  })
# ---
# name: test_sensors[sensor.technove_station_max_current-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_max_current',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.CURRENT: 'current'>,
    'original_icon': None,
    'original_name': 'Max current',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'max_current',
    'unique_id': 'AA:AA:AA:AA:AA:BB_max_current',
    'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
  })
# ---
# name: test_sensors[sensor.technove_station_max_current-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'current',
      'friendly_name': 'TechnoVE Station Max current',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_max_current',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '24',
  })
# ---
# name: test_sensors[sensor.technove_station_max_station_current-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_max_station_current',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.CURRENT: 'current'>,
    'original_icon': None,
    'original_name': 'Max station current',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'max_station_current',
    'unique_id': 'AA:AA:AA:AA:AA:BB_max_station_current',
    'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
  })
# ---
# name: test_sensors[sensor.technove_station_max_station_current-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'current',
      'friendly_name': 'TechnoVE Station Max station current',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfElectricCurrent.AMPERE: 'A'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_max_station_current',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '32',
  })
# ---
# name: test_sensors[sensor.technove_station_output_voltage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_output_voltage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.VOLTAGE: 'voltage'>,
    'original_icon': None,
    'original_name': 'Output voltage',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'voltage_out',
    'unique_id': 'AA:AA:AA:AA:AA:BB_voltage_out',
    'unit_of_measurement': <UnitOfElectricPotential.VOLT: 'V'>,
  })
# ---
# name: test_sensors[sensor.technove_station_output_voltage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'voltage',
      'friendly_name': 'TechnoVE Station Output voltage',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfElectricPotential.VOLT: 'V'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_output_voltage',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '238',
  })
# ---
# name: test_sensors[sensor.technove_station_signal_strength-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_signal_strength',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.SIGNAL_STRENGTH: 'signal_strength'>,
    'original_icon': None,
    'original_name': 'Signal strength',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'AA:AA:AA:AA:AA:BB_rssi',
    'unit_of_measurement': 'dBm',
  })
# ---
# name: test_sensors[sensor.technove_station_signal_strength-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'signal_strength',
      'friendly_name': 'TechnoVE Station Signal strength',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'dBm',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_signal_strength',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '-82',
  })
# ---
# name: test_sensors[sensor.technove_station_status-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'options': list([
        'unknown',
        'unplugged',
        'plugged_waiting',
        'plugged_charging',
      ]),
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_status',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.ENUM: 'enum'>,
    'original_icon': None,
    'original_name': 'Status',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'status',
    'unique_id': 'AA:AA:AA:AA:AA:BB_status',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[sensor.technove_station_status-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'enum',
      'friendly_name': 'TechnoVE Station Status',
      'options': list([
        'unknown',
        'unplugged',
        'plugged_waiting',
        'plugged_charging',
      ]),
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_status',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'plugged_charging',
  })
# ---
# name: test_sensors[sensor.technove_station_total_energy_usage-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.TOTAL: 'total'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_total_energy_usage',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.ENERGY: 'energy'>,
    'original_icon': None,
    'original_name': 'Total energy usage',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'energy_total',
    'unique_id': 'AA:AA:AA:AA:AA:BB_energy_total',
    'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
  })
# ---
# name: test_sensors[sensor.technove_station_total_energy_usage-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'energy',
      'friendly_name': 'TechnoVE Station Total energy usage',
      'state_class': <SensorStateClass.TOTAL: 'total'>,
      'unit_of_measurement': <UnitOfEnergy.KILO_WATT_HOUR: 'kWh'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_total_energy_usage',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '1234',
  })
# ---
# name: test_sensors[sensor.technove_station_wi_fi_network_name-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.technove_station_wi_fi_network_name',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:wifi',
    'original_name': 'Wi-Fi network name',
    'platform': 'technove',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'ssid',
    'unique_id': 'AA:AA:AA:AA:AA:BB_ssid',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensors[sensor.technove_station_wi_fi_network_name-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'TechnoVE Station Wi-Fi network name',
      'icon': 'mdi:wifi',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.technove_station_wi_fi_network_name',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'Connecting...',
  })
# ---
